### DISCUSSION TITLE
Stuck on "How to continue" - AIO interface missing & changing data directory

### DISCUSSION THREAD

**Discussion by viasux** [nextcloud/all-in-one#1257] [Questions] [answered]
I've followed the reverse proxy instructions to the "how-to-continue" step. From there, nothing seems to be progressing, as https://my-ip:8080 does not return anything.
https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#how-to-continue

The following is my ufw status:
```
Status: active

     To                         Action      From
     --                         ------      ----
[ 1] Anywhere                   ALLOW IN    192.168.0.0/24            
[ 2] 22/tcp                     LIMIT IN    Anywhere                  
[ 3] 80                         ALLOW IN    Anywhere                  
[ 4] 443                        ALLOW IN    Anywhere                  
[ 5] 8080/tcp                   ALLOW IN    Anywhere                  
[ 6] 8443/tcp                   ALLOW IN    Anywhere                  
[ 7] 3478/udp                   ALLOW IN    Anywhere                  
[ 8] 443/tcp                    ALLOW IN    Anywhere                  
[ 9] 80/tcp                     ALLOW IN    Anywhere                  
[10] 3478/tcp                   ALLOW IN    Anywhere                  
[11] 22/tcp (v6)                LIMIT IN    Anywhere (v6)             
[12] 80 (v6)                    ALLOW IN    Anywhere (v6)             
[13] 443 (v6)                   ALLOW IN    Anywhere (v6)             
[14] 8080/tcp (v6)              ALLOW IN    Anywhere (v6)             
[15] 8443/tcp (v6)              ALLOW IN    Anywhere (v6)             
[16] 3478/udp (v6)              ALLOW IN    Anywhere (v6)             
[17] 443/tcp (v6)               ALLOW IN    Anywhere (v6)             
[18] 80/tcp (v6)                ALLOW IN    Anywhere (v6)             
[19] 3478/tcp (v6)              ALLOW IN    Anywhere (v6)  
```

**Comment by viasux** [comment]
docker ps doesn't list the container, however running:
```
sudo docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 8080:8080 -e APACHE_PORT=11000 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest
```
returns:
```
docker: Error response from daemon: Conflict. The container name "/nextcloud-aio-mastercontainer" is already in use by container "5113bba1167e775a99705c510e92b02c2cdd2f1aa721b37d5798e4a81065d411". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.
```

**Reply by Zoey2936** [reply]
Try: `docker stop nextcloud-aio-mastercontainer && docker rm nextcloud-aio-mastercontainer`

**Reply by viasux** [reply]
That command appears to have worked, however the error persists:
```
docker: Error response from daemon: driver failed programming external connectivity on endpoint nextcloud-aio-mastercontainer (d33621d719baefa138d33cb8462670f192d750f55fd84ae0da3827503a854c72): Error starting userland proxy: listen tcp4 0.0.0.0:8080: bind: address already in use
```

**Reply by viasux** [reply]
wait no thats a different error lol

**Reply by Zoey2936** [reply]
Something on your server already uses port 8080

**Reply by viasux** [reply]
Should I change it to a different port then? :)

**Comment by Zoey2936** [comment, marked as answer]
change the part `--publish 8080:8080` to `--publish 8081:8080` then you should be able to open the aio panel on port 8081

**Reply by viasux** [reply]
Everything appears to be working! "Internect connection" though lmao
![image](https://user-images.githubusercontent.com/47131867/195692896-65dfe212-6b2f-4a29-b817-023a78813daa.png)

-- ill keep you updated if there's any more issues! Is there any way to donate to nextcloud?

**Reply by viasux** [reply]
And I ran into another problem. Step 3:
"Use this envorinmental variable during the initial startup of the mastercontainer to make the apache container only listen on localhost: -e APACHE_IP_BINDING=127.0.0.1. Attention: This is only recommended to be set if you use localhost in your reverse proxy config to connect to your AIO instance. If you use an ip-address, you can either simply skip this step or set it to 0.0.0.0 if you are unsure what the correct value is."

-- I don't know how to set that environmental variable - and is it already past when I was supposed to do so? I was just doing things in the order listed :p

**Reply by viasux** [reply]
Also, my terminal is stuck in a run command - that I'd have to control-c to get out of. is it safe to do so now?

**Reply by Zoey2936** [reply]
Open a new Terminal and run:
`docker stop nextcloud-aio-mastercontainer && docker rm nextcloud-aio-mastercontainer && docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 8081:8080 -e APACHE_PORT=11000 -e APACHE_IP_BINDING=127.0.0.1 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest`

**Reply by viasux** [reply]
alright, that fixed the first terminal (and im assuming the environment variable), the second terminal is now still in the command, is it safe to close it?

**Reply by Zoey2936** [reply]
Retry in a new windows this:
`docker stop nextcloud-aio-mastercontainer && docker rm nextcloud-aio-mastercontainer && docker run --sig-proxy=false --name nextcloud-aio-mastercontainer -d --restart always --publish 8081:8080 -e APACHE_PORT=11000 -e APACHE_IP_BINDING=127.0.0.1 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest`

**Reply by viasux** [reply]
Also, is there a way to change the data directory - before I put any important data onto the install?

**Reply by viasux** [reply]
the command worked for the terminal!

**Reply by Zoey2936** [reply]
> Also, is there a way to change the data directory - before I put any important data onto the install?

See: https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir

**Reply by viasux** [reply]
> > Also, is there a way to change the data directory - before I put any important data onto the install?
> 
> See: https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir

hm, i would have changed it before-hand had I known of a way, however I didn't see anything in the installation specifying an install path.. thank you though!

**Reply by Zoey2936** [reply]
If you want you can mark my comment as answer, if you want

**Reply by viasux** [reply]
This says "removing and recreating the mastercontainer with -e NEXTCLOUD_DATADIR="/mnt/ncdata""

what command am I supposed to set that option to?

**Reply by Zoey2936** [reply]
Just run:
`docker stop nextcloud-aio-mastercontainer && docker rm nextcloud-aio-mastercontainer && docker run --sig-proxy=false --name nextcloud-aio-mastercontainer -d --restart always --publish 8081:8080 -e APACHE_PORT=11000 -e APACHE_IP_BINDING=127.0.0.1 -e NEXTCLOUD_DATADIR="/mnt/ncdata" --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest` - but change /mnt/ncdata to the path you want

**Reply by viasux** [reply]
> Just run: `docker stop nextcloud-aio-mastercontainer && docker rm nextcloud-aio-mastercontainer && docker run --sig-proxy=false --name nextcloud-aio-mastercontainer -d --restart always --publish 8081:8080 -e APACHE_PORT=11000 -e APACHE_IP_BINDING=127.0.0.1 -e NEXTCLOUD_DATADIR="/mnt/ncdata" --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest` - but change /mnt/ncdata to the path you want

This appears to not be working. Is there a way to completely remove everything and start again? This time I'll set the environment variable and data directory from the start...

getting "Appdata is not present. Did you maybe change the datadir after aio was first started?"

**Reply by Zoey2936** [reply]
https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance
